R/checkGeneSymbols.R

```diff
@@ -0,0 +1,48 @@
+# checkGeneSymbols.R
+
+#' Check whether gene symbols given are valid HGNC gene symbols.
+#'
+#' \code{isGeneSymbol} Recieves a vector as input and checks whether the input
+#' vector contains valid HGNC gene symbols (case insensitive) by returning a
+#' vector of length input, containing TRUE for gene symbols and FALSE for others.
+#'
+#' @section Checking symbols
+#'   The function calls fmatch on each symbol given and checks the return value
+#'   of fmatch to see if it is a valid gene symbol. If fmatch returns a
+#'   value/index, the symbol must be a valid gene symbol; if fmatch returns
+#'   NA, the symbol must be invalid. The function uses is.na to check if the
+#'   return value of fmatch is NA.
+#'
+#' @param input A vector that must be checked for valid gene symbols.
+#' @return A vector of length input containing TRUE for valid HGNC
+#' gene symbols and FALSE for others.
+#'
+#'
+#' @seealso \code{\link{generateHGNCHash}} Generates a data frame containing
+#' all valid HGNC gene symbols, calls fmatch on the data frame so it will
+#' create a hash of the data frame to be used by checkGeneSymbols.R.
+#'
+#' @examples
+#' isGeneSymbol(c("123", "234"))
+#' isGeneSymbol(c("A1BG", "a1bg", "a1Bg", "A1bG))
+#' isGeneSymbol(c("123", "A1CF", "234", "a1bg"))
+
+isGeneSymbol <- function(input) {
+    # Check the validity of the input.
+    if (is.null(input) || length(input) == 0 || typeof(input) != "character" ||
+        mode(input) != "character" || class(input) != "character") {
+            stop("Invalid input!")
+    }
+
+    output = c()
+
+    # Check each component of the input vector.
```

Comment:
wow you are right I did not know you can fmatch a vector against a vector...

Here are the microbenchmark results:
```
> microbenchmark(isGeneSymbol(a), isGeneSymbol2(a))
Unit: microseconds
             expr      min         lq       mean    median        uq       max neval cld
  isGeneSymbol(a) 9792.464 10463.5370 11418.6488 10828.522 11501.450 16326.030   100   b
 isGeneSymbol2(a)  145.160   154.8995   194.1434   185.972   224.001   370.088   100  a 
```
isGeneSymbol is my version and isGeneSymbol2 is your version. Your version is much much faster!
Also, the input "a" included 1002 components.